File: src/main/kotlin/com/employeeDashboard/controller/MessageController.java
package com.employeeDashboard.controller;

import com.dashboard.controller.interfaces.EmployeeControllerInterface;
import com.dashboard.exception.*;
import com.dashboard.model.Employee;
import io.swagger.v3.oas.annotations.tags.Tag;
import lombok.extern.log4j.Log4j2;
import org.springframework.data.domain.Pageable;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/v1/messages")
@Tag(name = "Messages")
@Log4j2
public class MessageController implements EmployeeControllerInterface {

    @Override
    public ResponseEntity<?> createEmployee(Employee employee) {
        return new ResponseEntity<>(HttpStatus.NOT_IMPLEMENTED);
    }

    @Override
    public ResponseEntity<?> getAllEmployees(Pageable pageable) {
        return new ResponseEntity<>(HttpStatus.NOT_IMPLEMENTED);
    }

    @Override
    public ResponseEntity<?> getEmployeeById(Integer id) {
        return new ResponseEntity<>(HttpStatus.NOT_IMPLEMENTED);
    }

    @Override
    public ResponseEntity<?> deleteEmployeeById(Integer id) {
        return new ResponseEntity<>(HttpStatus.NOT_IMPLEMENTED);
    }

    @Override
    public ResponseEntity<?> updateEmployee(Integer employeeId, Employee employee) {
        return new ResponseEntity<>(HttpStatus.NOT_IMPLEMENTED);
    }

    @Override
    public ResponseEntity<?> assignTask(Integer employeeID, Integer taskID) {
        return new ResponseEntity<>(HttpStatus.NOT_IMPLEMENTED);
    }

    @Override
    public ResponseEntity<?> unassignTask(Integer employeeID, Integer taskID) {
        return new ResponseEntity<>(HttpStatus.NOT_IMPLEMENTED);
    }
}

File: src/main/kotlin/dashboard/api/MessageController.kt
package dashboard.api

import dashboard.interfaces.controllerInterfaces.MessageControllerInterface
import dashboard.models.jpa.Message
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.RestController
import org.springframework.web.bind.annotation.RequestMapping


@RestController
@RequestMapping("/api/v1/messages")
class MessageController : MessageControllerInterface {

    private val LOGGER: Logger = LoggerFactory.getLogger(MessageController::class.java)

    override fun createMessage(message: Message): ResponseEntity<*> {
        return ResponseEntity("NOT IMPLEMENTED", HttpStatus.NOT_IMPLEMENTED)
    }

    override fun allMessages(): ResponseEntity<*> = ResponseEntity("NOT IMPLEMENTED", HttpStatus.NOT_IMPLEMENTED)

    override fun getMessageById(messageID: Int): ResponseEntity<*> {
        return ResponseEntity("NOT IMPLEMENTED", HttpStatus.NOT_IMPLEMENTED)
    }

    override fun deleteMessageById(messageID: Int): ResponseEntity<*> {
        return ResponseEntity("NOT IMPLEMENTED", HttpStatus.NOT_IMPLEMENTED)
    }

    override fun updateMessage(messageID: Int, Message: Message): ResponseEntity<*> {
        return ResponseEntity("NOT IMPLEMENTED", HttpStatus.NOT_IMPLEMENTED)
    }
}